feat: extend toColor function by design tokens, update storybook and tests (#4910)
Sage/carbon: docs/colors.stories.mdx

| #582C83 | amethyst       | amethystTint       | amethystShade       | ❌           |
| #8246AF | plum           | plumTint           | plumShade           | ❌           |
| #003349 | slate          | slateTint          | slateShade          | ❌           |
| #000000 | ❌             | ❌                 | ❌                  | blackOpacity |
| #ffffff | ❌             | ❌                 | ❌                  | whiteOpacity |

Tint, Shade and Opacity properties must be followed by a number.

Where there is a collision between CSS strings and the palette color (e.g. gold) the palette color takes presedence.

```jsx
<Typography color='brilliantGreen'>...</Typography>
<Typography color='brilliantGreenTint20'>...</Typography>
<Typography color='brilliantGreenShade30'>...</Typography>
<Typography color='blackOpacity55'>...</Typography>
<Typography color='whiteOpacity22'>...</Typography>
```

## Design tokens

> Using design tokens ensures visual consistency across the application. If the designers update colors using a design tool, we have got always updated values.

Reviewer: ```suggestion
Using design tokens ensures visual consistency across the application and means that we will always have up-to-date values, if the colors are updated by designers using design tools
```

> The name of the chosen design token must be always preceded by `--`, for example, if we want to use `colorsYang100` token in our code, we have to pass `--colorsYang100` as a prop. See [Design Tokens](https://zeroheight.com/2ccf2b601/p/217e24-design-tokens/b/46fb17) documentation for more information.

Reviewer: This link leads to a doc page that is protected by a password. Not sure if it's necessary here tbh
